tests/test_aea_exectimeout.py

```diff
@@ -0,0 +1,178 @@
+
+    def tearDown(self) -> None:
+        """Tear down."""
+        self.aea_tool.stop()
+
+    def prepare(self, function: Callable) -> None:
+        """Prepare aea_tool for testing.
+
+        :param function: function be called on react handle or/and Behaviour.act
+        :return: None
+        """
+        agent_name = "MyAgent"
+
+        builder = AEABuilder()
+        builder.set_name(agent_name)
+        builder.add_private_key(FETCHAI, "")
+
+        self.function_finished = False
+
+        def handler_func(*args, **kwargs):
+            function()
+            self.function_finished = True
+
+        skill_context = SkillContext()
+        handler_cls = make_handler_cls_from_funcion(handler_func)
+
+        behaviour_cls = make_behaviour_cls_from_funcion(handler_func)
+
+        test_skill = Skill(
+            SkillConfig(name="test_skill"),
+            skill_context=skill_context,
+            handlers={
+                "handler1": handler_cls(name="handler1", skill_context=skill_context)
+            },
+            behaviours={
+                "behaviour1": behaviour_cls(
+                    name="behaviour1", skill_context=skill_context
+                )
+            },
+        )
+        skill_context._skill = test_skill  # weird hack
+
+        builder._add_component_to_resources(test_skill)
+        aea = builder.build()
+
+        self.aea_tool = AeaTool(aea)
+        self.aea_tool.put_inbox(AeaTool.dummy_envelope())
+
+    def test_long_handler_cancelled_by_timeout(self):
+        """Test long function terminated by timeout."""
+        num_sleeps = 10
+        sleep_time = 0.1
+        function_sleep_time = num_sleeps * sleep_time
+        execution_timeout = 0.5
+        assert execution_timeout < function_sleep_time
+
+        self.prepare(lambda: sleep_a_bit(sleep_time, num_sleeps))
+        self.aea_tool.set_execution_timeout(execution_timeout)
+        self.aea_tool.setup()
+
+        with timeit_context() as timeit:
+            self.aea_action()
+
+        assert execution_timeout <= timeit.time_passed <= function_sleep_time
+        assert not self.function_finished
+        self.aea_tool.stop()
+
+    def test_short_handler_not_cancelled_by_timeout(self):
+        """Test short function NOTterminated by timeout."""
+        num_sleeps = 1
+        sleep_time = 0.1
+        function_sleep_time = num_sleeps * sleep_time
+        execution_timeout = 0.5
+
+        assert function_sleep_time <= execution_timeout
+
+        self.prepare(lambda: sleep_a_bit(sleep_time, num_sleeps))
+        self.aea_tool.set_execution_timeout(execution_timeout)
+        self.aea_tool.setup()
+
+        with timeit_context() as timeit:
+            self.aea_action()
+
+        assert function_sleep_time <= timeit.time_passed <= execution_timeout
+        assert self.function_finished
+        self.aea_tool.stop()
+
+    def test_no_timeout(self):
+        """Test function NOT terminated by timeout cause timeout == 0."""
```

Comment:
Soo timout 0 is a special case? shouldn't timeout 0 mean everything is timed out?

Reply:
there is no meaning in set code already expired, it's easier to do not call it at all.